File: app/src/main/java/com/gallopdevs/athanhelper/views/DayViewAdapter.java
package com.gallopdevs.athanhelper.views;

import android.content.Context;
import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentPagerAdapter;
import android.util.Log;
import android.widget.Toast;

import java.util.Calendar;
import java.util.Observable;
import java.util.Observer;

/**
 * Created by jgell on 5/23/2017.
 */

public class DayViewAdapter extends FragmentPagerAdapter {

    private static final String TAG = "DayViewAdapter";

    private static final int NUM_ITEMS = 7;

    public DayViewAdapter(FragmentManager fm) {
        super(fm);
    }

    @Override
    public Fragment getItem(int i) {
        Fragment fragment = new DayViewFragment();
        Bundle bundle = new Bundle();
        Calendar c = Calendar.getInstance();
        int day = c.get(Calendar.DAY_OF_WEEK);
        bundle.putInt("day", day + i);
        bundle.putInt("count", i);
        fragment.setArguments(bundle);
        return fragment;
    }

    @Override
    public int getCount() {
        return NUM_ITEMS;
    }
}

File: app/src/main/java/com/gallopdevs/athanhelper/ui/dayview/DayViewAdapter.kt
package com.gallopdevs.athanhelper.ui.dayview

import android.os.Bundle
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentActivity
import androidx.viewpager2.adapter.FragmentStateAdapter

class DayViewAdapter(fragmentActivity: FragmentActivity) : FragmentStateAdapter(fragmentActivity) {

    override fun createFragment(i: Int): Fragment {
        val fragment = DayViewFragment()

        val bundle = Bundle().apply {
            putInt(PAGE_INDEX, i)
        }

        fragment.arguments = bundle
        return fragment
    }

    override fun getItemCount(): Int = NUM_ITEMS

    companion object {
        private const val NUM_ITEMS = 7
        const val PAGE_INDEX = "PAGE_INDEX"
    }
}
